update-project-version/action.yml

```
@@ -48,22 +48,40 @@ runs:
     run: python ${{ github.action_path }}/update_version.py ${{ inputs.version-rule }} ${{ inputs.use-dev-suffix == 'true' && '--dev' || '' }}
     shell: bash
     working-directory: ${{ inputs.project-directory }}
-  - name: Get changed files
-    id: get-changed-files
+  - name: Get PR details
+    id: get-pr-details
     run: |
-      echo "changed-files<<EOF" >> "$GITHUB_OUTPUT"
-      # Prefix with "- " to generate a Markdown list.
-      git diff --name-only | awk '{ print "-", $0 }' >> "$GITHUB_OUTPUT"
-      echo "EOF" >> "$GITHUB_OUTPUT"
-    shell: bash
+      import pathlib, os, subprocess
+      changed_files = [pathlib.Path(file) for file in subprocess.check_output(["git", "diff", "--name-only"], encoding="ascii").split()]
```

PSF recommends using `subprocess.run` at this point.

I think this is equivalent:
```suggestion
      changed_files = [pathlib.Path(file) for file in subprocess.run(["git", "diff", "--name-only"], encoding="ascii", capture_output=True, check=True, text=True).split()]
```

---

Ok, I'll update it.

subprocess.run returns a CompletedProcess object and I think text=True has a reasonable default encoding.